router/router_test.go

```diff
@@ -152,6 +152,12 @@ func initRouter() {
 	misc.Init()
 }
 
+type drainer struct{}
+
+func (d *drainer) Drain(job *jobsdb.JobT) (bool, string) {
+	return false, ""
+}
```

Comment:
Can we have a test where we have throttling enabled and jobs that should be drained that makes sure we drain right away? Making sure that `throttler.CheckLimitReached` isn't called could be a strategy.

```suggestion
type drainer struct {
	drain bool
}

func (d *drainer) Drain(job *jobsdb.JobT) (bool, string) {
	return d.drain, ""
}
```